app/grandchallenge/challenges/models.py

```diff
@@ -558,9 +559,67 @@ def remove_admin(self, user):
 
     @property
     def accepting_submissions(self):
```

Comment:
A general comment: this method is hard to follow with a lot of nested if statements. Can you think of a way to simplify it? Maybe split it out into some smaller methods?